### ISSUE TITLE
Deprecated functionality: strlen(): Passing null to parameter #1 - Abstract.php

### ISSUE THREAD

**Issue by ADDISON74** [OpenMage/magento-lts#4076]
I am getting this error with PHP 8.3 in the system.log file:

```Deprecated functionality: strlen(): Passing null to parameter #1 ($string) of type string is deprecated  in /var/www/html/app/code/core/Mage/Wishlist/Controller/Abstract.php on line 88```

**/app/code/core/Mage/Wishlist/Controller/Abstract.php on line 88**

In order to reproduce this issue log into a customer account in the Frontend then add any product to the Wishlist. You will be redirected to the customer account. A part of the URL is **wishlist/index/index/wishlist_id/[PRODUCT_ID]/**. Press the [Add All to the Cart] button. In my case I am getting an error message because the product has custom options and I have to choose them. If you take a look into the /var/log directory you will see a nice system.log file, having 371 kB as size and inside the same line 1745 times.

**Comment by ADDISON74** [comment]
Maybe I am tired after a long busy day but this error is really funny. Looking into that file there is not strlen() function used.

**Comment by ADDISON74** [comment]
Commenting the line in the file the error doesn't appear in the system.log anumore.

```
 $qtysString = $this->getRequest()->getParam('qty');
        if (isset($qtysString)) {
            //$qtys = array_filter(json_decode($qtysString), '\strlen');
        }
```

**Comment by ADDISON74** [comment]
Commenting the bellow line in the file the error doesn't appear in the system.log anymore.

```php
 $qtysString = $this->getRequest()->getParam('qty');
        if (isset($qtysString)) {
            //$qtys = array_filter(json_decode($qtysString), '\strlen');
        }
```

This proves there is an issue in that line. The **array_filter** function filters the values of an array using a callback function. In this case the callback function is **strlen**. 

By searching the code I did not find an implementation of this function in the OpenMage, except the one in this file ```/app/code/core/Mage/Core/Helper/String.php```.

```php
/**
     * Retrieve string length using default charset
     *
     * @param string $string
     * @return int
     */
    public function strlen($string)
    {
        return is_null($string) ? 0 : iconv_strlen($string, self::ICONV_CHARSET);
    }
```

**Comment by ADDISON74** [comment]
This is before Magento 1.8.1.0

```php
$qtys = $this->getRequest()->getParam('qty');
```

This is after Magento 1.8.1.0

```php
$qtysString = $this->getRequest()->getParam('qty');
$qtys =  array_filter(json_decode($qtysString), 'strlen');
```

This PR #2335 changed strlen to \strlen. I don't know what was its scope.

```php
$qtys = array_filter(json_decode($qtysString), '\strlen');
```

**Comment by ADDISON74** [comment]
Before PHP 8.1 this line would remove empty values from an array like null, false, '', but 0 and '0' will be kept. Starting with PHP 8.1 a deprecated error is issued as I mentioned in the initial report.  After doing a research on the Internet, there are a few implementations to solve this issue:

**This one is readable**
```php
$qtys = array_filter(json_decode($qtysString), function ($qtysString) { return strlen($qtysString ?? ''); });
```

**A shorter version**
```php
$qtys = array_filter(json_decode($qtysString), fn($qtysString) => strlen($qtysString ?? ''));
```

**Suppress the warning with @**
```php
$qtys = @array_filter(json_decode($qtysString), '\strlen');
```

I would like to get some feedback which implementation should be used.

Please note there is another file with this issue

https://github.com/OpenMage/magento-lts/blob/2b907ad6ed5edd66a419f7671beb6d47d2d8d6b5/app/code/core/Mage/ImportExport/Model/Import/Entity/Product.php#L1973C4-L1980C6

**Comment by fballiano** [comment]
arrow functions became available in php7.4 so all and all I think I like the shorter version :-)

**Comment by ADDISON74** [comment]
Fixed in #4083.
